lib/plugins/initAuth.js

```diff
@@ -1,51 +1,27 @@
-export default async ({ store, app }) => {
-  const options = <%= serialize(options) %>
+const options = <%= serialize(options) %>
+
+const { onAuthStateChangedAction, onAuthStateChangedMutation } = options
+
+export default async ({ store, app }, inject) => {
 
   // Set up a listener, mutations and action for every onAuthStateChanged by Firebase.
   // AND runs the functions once BEFORE the root Vue.js Application is instantiated.
-  await (async () => {
-    return await new Promise(resolve => {
-      app.$fireAuth.onAuthStateChanged(async authUser => {
-        if (authUser) {
-          try {
-            await _handleAuthSuccess(authUser)
-          } catch (e) {
-            await _handleAuthError(e)
-            Promise.reject(e)
-          }
-        }
-        resolve(authUser)
-      })
+
+  const unsubscribe = await new Promise(resolve => {
+    const unsubscribe = app.$fireAuth.onAuthStateChanged(async authUser => {
+      const claims = authUser ? (await authUser.getIdTokenResult()).claims : null
+
+      if (onAuthStateChangedMutation) {
+        store.commit(onStateChangedMutation, { authUser, claims })
```

Comment:
!!!
Should be on**Auth**StateChangedMutation

Reply:
Done (to be committed)